contracts/taskModules/ProxyModule.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.12;
+
+import {GelatoBytes} from "../vendor/gelato/GelatoBytes.sol";
+import {TaskModuleBase} from "./TaskModuleBase.sol";
+import {LibDataTypes} from "../libraries/LibDataTypes.sol";
+import {IOpsUserProxy} from "../interfaces/IOpsUserProxy.sol";
+import {IOpsUserProxyFactory} from "../interfaces/IOpsUserProxyFactory.sol";
+
+contract ProxyModule is TaskModuleBase {
+    using GelatoBytes for bytes;
+
+    IOpsUserProxyFactory public immutable opsUserProxyFactory;
+
+    constructor(IOpsUserProxyFactory _opsUserProxyFactory) {
+        opsUserProxyFactory = _opsUserProxyFactory;
+    }
+
+    /**
+     * @inheritdoc TaskModuleBase
+     * @dev _taskCreator cannot create task to other user's proxy
+     */
+    function onCreateTask(
+        bytes32,
+        address _taskCreator,
+        address _execAddress,
+        bytes calldata,
+        bytes calldata
+    ) external override {
+        address proxy = _deployIfNoProxy(_taskCreator);
```

Comment:
only the EOA can create tasks for its proxy right? We cannot create task with the proxy itself? 

Not sure if we will have any use cases that will require that. Can you think about any? @brandonchuah

Reply:
I think you can create task with the proxy for example if you want to use batch creation

Reply:
An idea is we could have the proxy be compliant with our relay system. Then we could do multichain task creations/transactions through the proxy. The task will be mapped to the owner of the proxy.